### PR TITLE
TMI2-512-code-quality

### PR DESCRIPTION
**Pull request by IlyasBaqqari-CabinetOffice**
[TMI2-512](https://technologyprogramme.atlassian.net/browse/TMI2-512?atlOrigin=eyJpIjoiZDBkMmJhODQ3NWI0NGNhOWIzOTg1MWYwZTM0ZDMwOTAiLCJwIjoiaiJ9)

As per the ticket, I have went through the code making improvements where possible.

The Project Error Analysis feature of IntelliJ flagged hundreds of warnings, however a large portion of these were false positives. I have narrowed down the genuine warnings wherever I can, implementing fixes such as:

- Removing unnecessary variables and imports
- Fixing typos
- Formatting improvements
- Other general improvements to the code

[TMI2-512]: https://technologyprogramme.atlassian.net/browse/TMI2-512?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/gov/cabinetoffice/api/config/S3ConfigProperties.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @Configuration("s3ConfigurationProperties")
-@ConfigurationProperties(prefix = "aws")
+@ConfigurationProperties(prefix = "s3")
```
Comment (GavCookCO) [inline src/main/java/gov/cabinetoffice/api/config/S3ConfigProperties.java:16]: if we don't change the application properties file to match this then the application will break

File: src/main/java/gov/cabinetoffice/api/repositories/SubmissionJDBCRepository.java [hunk 2]
```diff
@@ -42,9 +42,7 @@ public ApplicationListDTO getApplicationSubmissionsByFundingOrganisationId(final
         final SqlParameterSource applicationParameters = new MapSqlParameterSource()
                 .addValue("fundingOrgId", fundingOrgId);
 
-        final String query = new StringBuilder(APPLICATIONS_WITH_SUBMISSIONS_QUERY)
-                .append(AND_FUNDING_ORG_CLAUSE)
-                .toString();
+        final String query = APPLICATIONS_WITH_SUBMISSIONS_QUERY + AND_FUNDING_ORG_CLAUSE;
```
Comment (GavCookCO) [inline src/main/java/gov/cabinetoffice/api/repositories/SubmissionJDBCRepository.java:45]: Its a personal preference thing but I think this is more readable the other way

File: src/main/java/gov/cabinetoffice/api/config/S3ConfigProperties.java (hunk 1 above)
Comment (IlyasBaqqari-CabinetOffice) [reply src/main/java/gov/cabinetoffice/api/config/S3ConfigProperties.java:16]: Will revert this change.

File: src/main/java/gov/cabinetoffice/api/repositories/SubmissionJDBCRepository.java [hunk 3]
```diff
@@ -65,10 +63,7 @@ public ApplicationListDTO getApplicationSubmissionsByFundingOrganisationIdAndGgi
                 .addValue("fundingOrgId", fundingOrgId)
                 .addValue("ggisIdentifier", ggisIdentifier);
 
-        final String query = new StringBuilder(APPLICATIONS_WITH_SUBMISSIONS_QUERY)
-                .append(AND_FUNDING_ORG_CLAUSE)
-                .append(AND_GGIS_ID_CLAUSE)
-                .toString();
+        final String query = APPLICATIONS_WITH_SUBMISSIONS_QUERY + AND_FUNDING_ORG_CLAUSE + AND_GGIS_ID_CLAUSE;
```
[the remark this answers is not in the document]
Comment (IlyasBaqqari-CabinetOffice) [reply src/main/java/gov/cabinetoffice/api/repositories/SubmissionJDBCRepository.java:66]: Agreed, will revert this.

File: src/main/java/gov/cabinetoffice/api/repositories/SubmissionJDBCRepository.java (hunk 2 above)
Comment (IlyasBaqqari-CabinetOffice) [reply src/main/java/gov/cabinetoffice/api/repositories/SubmissionJDBCRepository.java:45]: Agreed, will revert this.

**Review by jimpurvisTCO — APPROVED** [review]

**Pull request merged**